src/main/java/bsh/ClassGeneratorUtil.java

```diff
@@ -622,24 +622,35 @@ private void doSwitchBranch(int index, String targetClassName, String[] paramTyp
         // Unload the arguments from the ConstructorArgs object
         for (String type : paramTypes) {
             final String method;
-            if (type.equals("Z"))
+            switch ( type ) {
+            case "Z":
                 method = "getBoolean";
-            else if (type.equals("B"))
+                break;
+            case "B":
                 method = "getByte";
-            else if (type.equals("C"))
+                break;
+            case "C":
                 method = "getChar";
-            else if (type.equals("S"))
+                break;
+            case "S":
                 method = "getShort";
-            else if (type.equals("I"))
+                break;
+            case "I":
                 method = "getInt";
-            else if (type.equals("J"))
+                break;
+            case "J":
                 method = "getLong";
-            else if (type.equals("D"))
+                break;
+            case "D":
                 method = "getDouble";
-            else if (type.equals("F"))
+                break;
+            case "F":
                 method = "getFloat";
-            else
+                break;
+            default:
                 method = "getObject";
+                break;
+            }
```

Comment:
Nice!  I did some benchmarks on this and switch statements really are about 75% quicker even on simple decisions like:

```java
if ( a == 1 || a == 2 )
       // do it
```

is slower than 

```java
switch (a) {
   case 1: case 2:
       // do it
}
```